### PR TITLE
Airport Challenge - end of Week 1

### PR DESCRIPTION
**Pull request by samanthaixer**
This is my weekend Airport Challenge for Week 1

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,35 @@
...
 =================
 
+My approach
+---------
+
+* I read the user stories and broke them down into things I need the programme to do
+* For each line of "I want to be able to do this", I wrote the simplest test, fixed it, and repeated. I refactored my code every time the test passed.
+* Every now and again I refactored tests to make them easier to read/find and to ensure they didn't do unnecessary steps
+* Every now and again I ran rubocop to lint my code
+* Once I had done enough tests to complete a particular thing the code should do, I check the user stories to make sure I was still following the brief
```
Comment (Hives) [inline README.md]: sounds good!

File: README.md [hunk 2]
```diff
@@ -1,6 +1,35 @@
...
+1. I didn't feature test anything in ```irb``` as I went. Would this have changed my design/approach at all?
+2. I have written what I think is the cleanest, simplest code for the task that I have ever written. My concern - have I actually answered all of the brief?
+3. My head just doesn't want to think up new edge cases!
+
+How to run my code
+-------
+
+1. Run any of the test files to test the classes. ```airport_spec``` or ```weather_spec``` or ```plane_spec```
+2. Run any of the commands in ```irb``` or a ruby file ```Airport.new``` ```Plane.new``` ```Weather.new```
+
```
Comment (Hives) [inline README.md]: i think you need to include something like:
- run `gem install bundle` if you haven't already
- run `bundle`
or else the user won't be able to get the tests to run?

i think you should also give more explicit instructions about how to run the program, e.g.:
- open irb
- run `airport = Airport.new`
- run `plane = Plane.new`
- run `airport.land plane`
etc. (i haven't read any of your code yet, so i don't know how you've implemented it :) )

File: README.md [hunk 3]
```diff
@@ -1,6 +1,35 @@
...
+* NOTE: Line 7 of the weather class is not covered by a test. I have not worked out how to cover it.
+* NOTE: I have started the feature test but haven't worked out how to make it work with RSpec feature yet.
+
+Reflection
+-------
+
+1. I didn't feature test anything in ```irb``` as I went. Would this have changed my design/approach at all?
+2. I have written what I think is the cleanest, simplest code for the task that I have ever written. My concern - have I actually answered all of the brief?
+3. My head just doesn't want to think up new edge cases!
```
Comment (Hives) [inline README.md]: i feel your pain

File: README.md [hunk 4]
```diff
@@ -1,6 +1,35 @@
 Airport Challenge
 =================
```
Comment (Hives) [inline README.md]: code review checklist says you should have a description of what the program does at the top of the README

File: lib/Plane.rb [hunk 5]
```diff
@@ -0,0 +1,2 @@
+class Plane
+end
```
Comment (Hives) [inline lib/Plane.rb]: title of source files should be in snake_case (i.e. `plane.rb`). (rubocop picks this up)

File: lib/Airport.rb [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+      return "Successful take off"
+    end
+
+    raise "That plane isn't at the airport"
+  end
+
+  def capacity
+    return @max_capacity - @plane_list.length
+  end
+end
```
Comment (Hives) [inline lib/Airport.rb]: title of source files should be in snake_case (i.e. `airport.rb`). (rubocop picks this up)

File: spec/airport_spec.rb [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+      allow(subject).to receive(:current_weather) { "Sunny" }
+      expect(subject.land(Plane.new)).to eq "Plane landed"
+    end
+
+    it 'should stop a plane from landing when the weather is Stormy' do
+      allow(subject).to receive(:current_weather) { "Stormy" }
+      expect { subject.land(Plane.new) }.to raise_error "No landing allowed - Stormy weather"
+    end
+
+    it 'should stop a plan from landing when the airport is full' do
```
Comment (Hives) [inline spec/airport_spec.rb]: 'plan' -> 'plane'

File: spec/airport_spec.rb [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+    it { is_expected.to respond_to(:take_off).with(1).argument }
+
+    it 'should return a confirmation that the plane has taken off' do
+      allow(subject).to receive(:current_weather) { "Sunny" }
+      my_plane = Plane.new
+      subject.land(my_plane)
+      expect(subject.take_off(my_plane)).to eq "Successful take off"
+    end
+
+    it 'should raise an error if I try to make a non-existent plane take off' do
```
Comment (Hives) [inline spec/airport_spec.rb]: the plane isn't non-existent, it's just not at the airport. you could make the description of this test more explicit.

File: lib/weather.rb [hunk 9]
```diff
@@ -0,0 +1,9 @@
+class Weather
+  def current_weather
+    random = Random.rand(1..50)
+    p random
```
Comment (Hives) [inline lib/weather.rb]: can delete this line

File: lib/weather.rb [hunk 10]
```diff
@@ -0,0 +1,9 @@
+class Weather
+  def current_weather
```
Comment (Hives) [inline lib/weather.rb]: if you renamed this method something like `stormy?` you coud have it return either true or false, which would allow you to write it more briefly. You'd have to rewrite the relevant parts of airport.rb too

File: lib/Airport.rb [hunk 11]
```diff
@@ -0,0 +1,35 @@
+class Airport
+
+  attr_reader :plane_list
+  attr_reader :max_capacity
+
+  def initialize(max_capacity = 20)
```
Comment (Hives) [inline lib/Airport.rb]: 20 is a magic number here. refactor using a constant.

File: lib/Airport.rb [hunk 12]
```diff
@@ -0,0 +1,35 @@
+class Airport
+
+  attr_reader :plane_list
+  attr_reader :max_capacity
```
Comment (Hives) [inline lib/Airport.rb]: i think both of these variables should be private...

do you actually need to access max_capacity from outside the object, since you've got the `capactity` method?

[the rubric for the exercise says the plane list should be private](https://github.com/makersacademy/airport_challenge/blob/master/docs/review.md#do-not-expose-internal-implementation), so you could make a `airport.include? plane` method which would allow you to check whether a particular plane is at the airport

File: lib/Airport.rb [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+  def take_off(plane)
+    raise "No take-off allowed - Stormy weather" if current_weather == "Stormy"
+    raise "There are no planes at the airport" if plane_list.empty?
+
+    if @plane_list.include?(plane)
+      @plane_list.delete(plane)
+      return "Successful take off"
+    end
+
+    raise "That plane isn't at the airport"
```
Comment (Hives) [inline lib/Airport.rb]: guard condition should go at the top of the method i think?

File: README.md (hunk 4 above)
Comment (samanthaixer) [reply README.md]: Good point!

File: README.md (hunk 2 above)
Comment (samanthaixer) [reply README.md]: Agreed

File: lib/Airport.rb (hunk 12 above)
Comment (samanthaixer) [reply lib/Airport.rb]: Good to know. I would change this

File: lib/Airport.rb (hunk 13 above)
Comment (samanthaixer) [reply lib/Airport.rb]: Sounds good

File: lib/Airport.rb (hunk 11 above)
Comment (samanthaixer) [reply lib/Airport.rb]: I have made this change, good call

File: lib/Airport.rb (hunk 6 above)
Comment (samanthaixer) [reply lib/Airport.rb]: On my machine all of the source files are in snake_case, not sure why this is showing with a capital on git! Will need to have a play with adding and removing files maybe
